### DISCUSSION TITLE
supabase.auth.getUser() is too slow and supabase.auth.getSession() is insecure

### DISCUSSION THREAD

**Discussion by 2020-byte** [supabase/supabase#28983] [Questions] [answered]
What is the best practice for using middleware to allow access only to authenticated users, given that supabase.auth.getUser() is too slow and supabase.auth.getSession() is insecure? I'm looking for a solution that is commonly used, fast, and secure.

**Comment by X-Titouan** [comment, marked as answer]
I have the same question, but it seems that the right people are working on it:

https://github.com/orgs/supabase/discussions/27037
https://github.com/supabase/auth-js/issues/898

For now it seems that you have to choose between security and performances :/
